client/src/components/Query/Statement.jsx

```diff
@@ -180,15 +180,25 @@ class Statement extends React.Component {
   }
 
   handleDuplicate(query) {
+
     if (this.isDuplicatable()) {
-      const { draft } = this.state;
+      const { draft , display } = this.state;
+
       this.commit(draft);
       const index = query.index + 1;
       const clone = cloneDeep(query);
       clone.data.key = uuidv1();
+
       draft.splice(index, 0, clone.data);
+
+      const displayClone = cloneDeep(display);
```

Comment:
@claudia1296 Bonus points for knowing difference between shallow copy and deep copy in JS? :)

Reply:
**copy:** Copie exactement l'objet et prend les références ,donc l'original peut-être affecté
**Deep Copy :**  copie en profondeur, des nouvelles valeurs sont créées il ne touchera jamais à l'original.